presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -820,7 +820,7 @@ private static Table buildTableObject(
             Optional<HiveBucketProperty> bucketProperty,
             Map<String, String> additionalTableParameters,
             Path targetPath,
-            boolean external,
+            String tableType,
             String prestoVersion)
     {
         Map<String, HiveColumnHandle> columnHandlesByName = Maps.uniqueIndex(columnHandles, HiveColumnHandle::getName);
```

Comment:
nit: Do we want to check `tableType` has to be either `EXTERNAL_TABLE` or `MANAGED_TABLE` ?

Reply:
It can also be `TEMPORARY_TABLE`.